pandas/_libs/pd_parser.c

```diff
@@ -0,0 +1,270 @@
+  } else if (PyUnicode_Check(str)) {
+    tmp = PyUnicode_AsUTF8String(str);
+    if (tmp == NULL) {
+      return -1;
+    }
+    data = PyBytes_AS_STRING(tmp);
+  } else {
+    PyErr_SetString(PyExc_TypeError, "Invalid object type");
+    return -1;
+  }
+
+  status = pandas_to_double(data, result, sci, dec, maybe_int);
+
+  if (!status) {
+    /* handle inf/-inf infinity/-infinity */
+    if (strlen(data) == 3) {
+      if (0 == strcasecmp(data, "inf")) {
+        *result = HUGE_VAL;
+        *maybe_int = 0;
+      } else {
+        goto parsingerror;
+      }
+    } else if (strlen(data) == 4) {
+      if (0 == strcasecmp(data, "-inf")) {
+        *result = -HUGE_VAL;
+        *maybe_int = 0;
+      } else if (0 == strcasecmp(data, "+inf")) {
+        *result = HUGE_VAL;
+        *maybe_int = 0;
+      } else {
+        goto parsingerror;
+      }
+    } else if (strlen(data) == 8) {
+      if (0 == strcasecmp(data, "infinity")) {
+        *result = HUGE_VAL;
+        *maybe_int = 0;
+      } else {
+        goto parsingerror;
+      }
+    } else if (strlen(data) == 9) {
+      if (0 == strcasecmp(data, "-infinity")) {
+        *result = -HUGE_VAL;
+        *maybe_int = 0;
+      } else if (0 == strcasecmp(data, "+infinity")) {
+        *result = HUGE_VAL;
+        *maybe_int = 0;
+      } else {
+        goto parsingerror;
+      }
+    } else {
+      goto parsingerror;
+    }
+  }
+
+  Py_XDECREF(tmp);
+  return 0;
+
+parsingerror:
+  PyErr_Format(PyExc_ValueError, "Unable to parse string \"%s\"", data);
+  Py_XDECREF(tmp);
+  return -1;
+}
+
+static void *pandas_new_rd_source(PyObject *obj) { return new_rd_source(obj); }
+
+static int pandas_del_rd_source(void *src) { return del_rd_source(src); }
+
+static void *pandas_buffer_rd_bytes(void *source, size_t nbytes,
+                                    size_t *bytes_read, int *status,
+                                    const char *encoding_errors) {
+  return buffer_rd_bytes(source, nbytes, bytes_read, status, encoding_errors);
+}
+
+static void pandas_uint_state_init(uint_state *self) { uint_state_init(self); }
+
+static int pandas_uint64_conflict(uint_state *self) {
+  return uint64_conflict(self);
+}
+
+static void pandas_coliter_setup(coliter_t *self, parser_t *parser, int64_t i,
+                                 int64_t start) {
+  coliter_setup(self, parser, i, start);
+}
+
+static parser_t *pandas_parser_new(void) { return parser_new(); }
+
+static int pandas_parser_init(parser_t *self) { return parser_init(self); }
+
+static void pandas_parser_free(parser_t *self) { parser_free(self); }
+
+static void pandas_parser_del(parser_t *self) { parser_del(self); }
+
+static int pandas_parser_add_skiprow(parser_t *self, int64_t row) {
+  return parser_add_skiprow(self, row);
+}
+
+static int pandas_parser_set_skipfirstnrows(parser_t *self, int64_t nrows) {
+  return parser_set_skipfirstnrows(self, nrows);
+}
+
+static void pandas_parser_set_default_options(parser_t *self) {
```

Comment:
for my edification, why is everything pandas_foo?  is it bc c doesn't have namespaces so this avoids potential overlap?

Reply:
Yea correct - there is no name spacing so don't want to redefine these symbols

Reply:
These might be extraneous - let me see if we can reference the function directly